build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/compat/compat/RestCompatTestTransformTask.java

```diff
@@ -90,30 +91,28 @@ public class RestCompatTestTransformTask extends DefaultTask {
     private final DirectoryProperty sourceDirectory;
     private final DirectoryProperty outputDirectory;
     private final PatternFilterable testPatternSet;
-    private final Factory<PatternSet> patternSetFactory;
-    private final List<RestTestTransform<?>> transformations = new ArrayList<>();
     // PatternFilterable -> reason why skipped.
     private final Map<PatternFilterable, String> skippedTestByFilePatternTransformations = new HashMap<>();
     // PatternFilterable -> list of full test names and reasons. Needed for 1 pattern may include many tests and reasons
     private final Map<PatternFilterable, List<Pair<String, String>>> skippedTestByTestNameTransformations = new HashMap<>();
 
     @Inject
-    public RestCompatTestTransformTask(
-        FileSystemOperations fileSystemOperations,
-        Factory<PatternSet> patternSetFactory,
-        ObjectFactory objectFactory
-    ) {
-        this.patternSetFactory = patternSetFactory;
+    protected Factory<PatternSet> getPatternSetFactory() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Inject
+    public RestCompatTestTransformTask(FileSystemOperations fileSystemOperations, ObjectFactory objectFactory) {
         this.fileSystemOperations = fileSystemOperations;
         this.compatibleVersion = Version.fromString(VersionProperties.getVersions().get("elasticsearch")).getMajor() - 1;
         this.sourceDirectory = objectFactory.directoryProperty();
         this.outputDirectory = objectFactory.directoryProperty();
-        this.testPatternSet = patternSetFactory.create();
+        this.testPatternSet = getPatternSetFactory().create();
         this.testPatternSet.include("/*" + "*/*.yml"); // concat these strings to keep build from thinking this is invalid javadoc
         // always inject compat headers
         headers.put("Content-Type", "application/vnd.elasticsearch+json;compatible-with=" + compatibleVersion);
         headers.put("Accept", "application/vnd.elasticsearch+json;compatible-with=" + compatibleVersion);
-        transformations.add(new InjectHeaders(headers, Set.of(RestCompatTestTransformTask::doesNotHaveCatOperation)));
+        getTransformations().add(new InjectHeaders(headers, Sets.newHashSet(RestCompatTestTransformTask::doesNotHaveCatOperation)));
```

Comment:
refactor to use best practices and get rid of property and use `abstract  ListProperty getTransformations()` instead